bin/compile

```diff
@@ -440,7 +445,14 @@ case "${TOOL}" in
     gomodules)
         pkgs=${GO_INSTALL_PACKAGE_SPEC:-$(awk '{ if ($1 == "//" && $2 == "+heroku" && $3 == "install" ) { print substr($0, index($0,$4)); exit }}' ${goMOD})}
         if [ -z "${pkgs}" ]; then
-            pkgs="default"
+            if [ -d "${build}/cmd" ]; then
+                step "Detected \`cmd\` folder. Installing packages under \`./cmd\`"
+                step "To install packages under MyDir/ add the following to \`go.mod\` file:"
```

Comment:
> "Detected \`cmd\` folder. Installing packages under \`./cmd\`"

- The backticks are unnecessary when viewed in the terminal
- I would prefer to not have the `./` as not everyone may be familiar with that unix convention. Perhaps we could list the different "applications" we're installing dependencies for?

Ex:

```
Detected "cmd" folder. Installing packages for:
  cmd/web
  cmd/migrate
  cmd/worker
```

Reply:
> step "To install packages under MyDir/ add the following to \`go.mod\` file:"
> warn "// +heroku install ./MyDir/..."

Do we need this warning? It will be displayed for every build

Reply:
lets chat about this.

Reply:
`./cmd/...` is a go thing, inspired by the unix convention, so if you're writting go, you'll run into it and likely be familiar with it.

Reply:
PS: Don't confuse a package spec with a directory, they're not the same thing, even though packages are contained within directories.

For instance, give a package named `github.com/foo/bar`, if the buildpack were building that code and it contained the `cmd/foo` and `cmd/bar` directories, both with main package in, then the package spec of `./cmd/...` and `github.com/foo/bar/cmd/...` are essentially equivalent.

Reply:
With that said, it would be nice to detail what package spec we've decided AND what commands we're installing because of it.